frontend/imports/components/startWatchingDialog.tsx

```diff
@@ -0,0 +1,113 @@
+/** @jsxImportSource @emotion/react */
+import React, { ChangeEvent, useState } from 'react'
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Button,
+  TextField,
+  Typography
+} from '@mui/material'
+import config from '../../config.json'
+import { useRouter } from 'next/router'
+import { css } from '@emotion/react'
+
+const onEnter = (func: () => void | Promise<void>) => (e: React.KeyboardEvent<HTMLDivElement>) => {
+  if (e.key === 'Enter') return func()
+}
+
+const StartWatchingDialog = (props: { shown: boolean, handleClose: () => void }) => {
+  const [title, setTitle] = useState('')
+  const [fileName, setFileName] = useState('')
+  const [error, setError] = useState('')
+  const [inProgress, setInProgress] = useState(false)
+
+  const router = useRouter()
+
+  const handleClose = () => {
+    props.handleClose()
+  }
+
+  const handleCreateRoom = async () => {
+    setInProgress(true)
+    try {
+      const req = await fetch(config.serverUrl + '/api/room', {
+        method: 'POST',
+        body: JSON.stringify({ title })
+      })
+      const res: { error?: string, id: string } = await req.json()
+      if (res.error) setError(res.error)
+      else {
+        props.handleClose()
+        router.push(`/room/${res.id}`).catch(console.error)
+      }
+    } catch (e) { setError('An unknown network error occurred.') }
+  }
+
+  const handleFileUpload = (e: ChangeEvent<HTMLInputElement>) => {
+    if (!e.target.files) {
```

Comment:
Check for the length, avoid any dumb problems.
```suggestion
    if (e.target.files?.length !== 1) {
```